### ISSUE TITLE
Password length is locked between [6, 32]. Avatar URL length is chopped off after 64 characters.

### ISSUE THREAD

**Issue by lylythechosenone** [CSP02/ChatterBoxAPI#5]
```js
router.post("/signup", [
    check('username').trim().isLength({ min: 3, max: 32 }),
    check('password').isLength({ min: 6, max: 32 }),
    check('color').isHexColor()
]
```
This is especially strange, because required password length is said to be *at least 8 characters*, with no maximum.
```js
        const user = new User({
            username: username,
            password: password,
            color: color.slice(0, 6),
            avatarURL: avatarURL.slice(0, 64)
        });
```
This could chop up a valid url.

**Comment by CSP02** [comment]
well the reason i limited password and avatarURL is because i dont want my db to fill (at least not yet). and you can use a url shortner for the avatarURL.

**Comment by lylythechosenone** [comment]
True, but these limits are not stated anywhere. Also, the avatar URL one doesn't even fail - it just uses a chopped off URL. For passwords, I would recommend a limit of at least 128 characters. 32 characters of valid UTF-8 only allows 224 bits (28 bytes) of entropy, which may be insufficient.

**Comment by CSP02** [comment]
oh yeah i forcing it xd
will let the users know about it when they are signing up and updating their profiles
